packages/components/src/Button/MultiFunctionButton.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2021 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, {
+  FC,
+  ReactElement,
+  useRef,
+  cloneElement,
+  useEffect,
+  useState,
+  RefObject,
+} from 'react'
+import styled from 'styled-components'
+import { ButtonProps, IconButtonProps } from '.'
+
+export interface MultiFunctionButtonProps {
+  alternate: ReactElement<
+    (ButtonProps | IconButtonProps) & {
+      ref: RefObject<HTMLButtonElement>
+      ariaHidden: boolean
+    }
+  >
+  children: ReactElement<ButtonProps | IconButtonProps>
+  setChildrenRef?: any
+  setAlternateRef?: any
+  swap?: boolean
+}
+
+export const MultiFunctionButton: FC<MultiFunctionButtonProps> = ({
+  alternate,
+  children,
+  setChildrenRef,
+  setAlternateRef,
+  swap = false,
+}) => {
```

Comment:
This will require `displayName` – see suggestion below.
```suggestion
export const MultiFunctionButton = forwardRef(
  (
    {
      alternate,
      children,
      alternateRef,
      swap = false,
    }: MultiFunctionButtonProps,
    forwardedRef: Ref<HTMLButtonElement>
  ) => {
```